packages/block-editor/src/hooks/layout.js

```diff
@@ -66,23 +49,24 @@ function LayoutPanel( { setAttributes, attributes, name: blockName } ) {
 		return null;
 	}
 
-	const { allowLayoutSwitching, defaultBlockLayout } = getLayoutBlockSettings(
-		blockName
-	);
+	const {
+		allowSwitching: canBlockSwitchLayout,
+		default: defaultBlockLayout,
+	} = getBlockSupport( blockName, layoutBlockSupportKey ) || {};
 
 	const usedLayout = layout ? layout : defaultBlockLayout || {};
 	const { inherit = false, type = 'default' } = usedLayout;
 	const layoutType = getLayoutType( type );
 
 	const onChangeType = ( newType ) =>
-		setAttributes( { layout: { type: newType } } );
+		setAttributes( { layout: { type: newType } } ); // TODO needs checking with block defaults.
```

Comment:
Can you clarify this comment?

Reply:
I'm not sure yet 😄 . I'm adding some comments to check it more when the time comes. I just thought I should check if we would need to add the `defaultBlockLayout` props if exist as well, because this line will trigger a rerender and the logic is if `layout` is truthy. Therefore I'm not sure it will behave properly.